msi/msi_test.go

```diff
@@ -70,7 +70,91 @@ func TestCanGetMsi(t *testing.T) {
 	if err != nil {
 		t.Fatal(err.Error())
 	}
-	err = ioutil.WriteFile(fmt.Sprintf("%s/msi.json", outdir), msiJsonBytes[:], 0700)
+	err = ioutil.WriteFile(fmt.Sprintf("%s/msi1.json", outdir), msiJsonBytes[:], 0700)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+}
+
+func TestCanGetMsiForStorage(t *testing.T) {
+	t.Skip() // for testing on Azure VM only
+	outdir := "./testoutput"
+	os.Mkdir(outdir, 0777)
+	secureHttpClient := httputil.NewSecureHttpClient(httputil.NoRetry)
+	msiProvider := NewMsiProvider(secureHttpClient)
+	msi, err := msiProvider.GetMsiForResoruce("https://storage.azure.com/")
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	t.Logf("Successfully got msi token.\nClientId was : %s", msi.ClientID)
+	msiJsonBytes, err := json.Marshal(msi)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	err = ioutil.WriteFile(fmt.Sprintf("%s/msi2.json", outdir), msiJsonBytes[:], 0700)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+}
+
+func TestCanGetMsiForKeyVault(t *testing.T) {
+	t.Skip() // for testing on Azure VM only
+	outdir := "./testoutput"
+	os.Mkdir(outdir, 0777)
+	secureHttpClient := httputil.NewSecureHttpClient(httputil.NoRetry)
+	msiProvider := NewMsiProvider(secureHttpClient)
+	msi, err := msiProvider.GetMsiForResoruce("https://vault.azure.net")
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	t.Logf("Successfully got msi token.\nClientId was : %s", msi.ClientID)
+	msiJsonBytes, err := json.Marshal(msi)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	err = ioutil.WriteFile(fmt.Sprintf("%s/msi3.json", outdir), msiJsonBytes[:], 0700)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+}
+
+func TestCanGetMsiForKeyVaultWithClientId(t *testing.T) {
+	t.Skip() // for testing on Azure VM only
+	outdir := "./testoutput"
+	os.Mkdir(outdir, 0777)
+	secureHttpClient := httputil.NewSecureHttpClient(httputil.NoRetry)
+	msiProvider := NewMsiProvider(secureHttpClient)
+	msi, err := msiProvider.GetMsiUsingClientId("31b403aa-c364-4240-a7ff-d85fb6cd7232", "https://vault.azure.net")
```

Comment:
where is this clientId from?

Reply:
added a comment explaining were this value comes from. Note that the test is marked as t.skip(), its used for manual verification as of now. The infrastructure doesn't exist in this repo yet.